app/src/main/java/ca/rmen/android/poetassistant/main/dictionaries/Search.java

```diff
@@ -182,39 +176,13 @@ protected void onPostExecute(@Nullable String word) {
     }
 
     public void addSuggestion(String query) {
-        mSuggestionsAdapter.addSuggestion(query);
+        ContentValues contentValues = new ContentValues(1);
+        contentValues.put(SearchManager.QUERY, query);
+        mSearchableActivity.getContentResolver().insert(SuggestionsProvider.CONTENT_URI, contentValues);
     }
 
     public void clearSearchHistory() {
-        mSuggestionsAdapter.clear();
+        mSearchableActivity.getContentResolver().delete(SuggestionsProvider.CONTENT_URI, null, null);
     }
 
-    private final SearchView.OnQueryTextListener mOnQueryTextListener = new SearchView.OnQueryTextListener() {
-        @Override
-        public boolean onQueryTextSubmit(String query) {
-            return false;
-        }
-
-        @Override
-        public boolean onQueryTextChange(String newText) {
-            mSuggestionsAdapter.filterSuggestions(newText);
-            return false;
-        }
-    };
-
-    private final SearchView.OnSuggestionListener mOnSuggestionListener = new SearchView.OnSuggestionListener() {
-        @Override
-        public boolean onSuggestionSelect(int position) {
-            String suggestion = mSuggestionsAdapter.getSuggestion(position);
-            mSearchView.setQuery(suggestion, false);
-            return false;
-        }
-
-        @Override
-        public boolean onSuggestionClick(int position) {
-            String suggestion = mSuggestionsAdapter.getSuggestion(position);
-            mSearchView.setQuery(suggestion, true);
-            return true;
-        }
-    };
```

Comment:
Pro: don't need to handle updating the suggestions when the user types, or setting the query when the user clicks on a suggestion.